_posts/2017-05-08-refactoring-identities.md

```diff
@@ -0,0 +1,50 @@
+
+### Why it didn't work
+
+In retrospect, it was a mistake to try to solve all of these problems at the same time, as they are all fundamentally different. By trying to link them together, we largely failed to solve any of them.
+
+The idea that your Google and Github accounts each represented a different "identity" confused many users. Most people in fact only have one persona, and their accounts on various services represent that same persona. But if you attached both to your Sandstorm account, then Sandstorm would begin asking you to choose which identity you wished to act as -- a question that made no sense to most people. Moreover, other people, when sharing with you, would begin to see multiple "copies" of you showing up in their contacts list -- one for each credential. Often it was hard for people to tell what the difference was or which they should choose. Because of this confusion, the Sandstorm team has tended to steer people away from linking multiple identities in the first place. Obviously, this defeats the goals for which multi-identity was created.
+
+Meanwhile, for people who actually wanted to manage multiple personas -- pseudonyms, brand identities, etc. -- we never managed to provide a good experience. It was very easy to accidentally use the wrong identity and reveal yourself. At the same time, the need to support this use case often exacerbated the confusion for users who only had one logical persona (as described in the previous paragraph). Without this design constraint, we could make a better UX for most people.
+
+Finally, in practice this design didn't really solve the ability to move grains between Sandstorm servers, because different Sandstorm servers may very well use totally different login mechanisms. Often, people want to transition grains from Oasis -- where they used e-mail login -- to a new self-hosted server -- where they used LDAP or SAML login. Since the login mechanisms differed, user IDs didn't match up anyway. To really solve this problem, we need something more dynamic.
+
+### What we're doing instead
+
+We are dropping support for multiple "personas". Going forward, each Sandstorm account will have only one profile -- one name, one profile picture, one entry in the sharing auto-complete list, etc. What we call "identities" today will be renamed to "credentials". A credential will no longer have its own profile, and you will no longer need to choose which credential you are acting as when accessing grains. Authentication-related features of credentials (for login, and for secure sharing) will remain mostly intact.
```

Comment:
Does this mean all existing identities will be merged into one identity (and one account) or that each identity will be split into separate accounts?